cou/apps/app.py

```diff
@@ -338,6 +341,44 @@ async def _check_upgrade(self, target: OpenStackRelease) -> None:
             )
             raise ApplicationError()
 
+    async def _run_package_upgrade_command(self, unit: str, command: str) -> None:
+        """Run package upgrade command and raise Exception if fails.
+
+        :param unit: The name of the unit where the command runs on.
+        :type str
+        :param command: The upgrade command to run.
+        :type str
+        :raises PackageUpgradeError: When the upgrade command execution resulted in error.
+        """
+        logger.info("Running '%s' on %s", command, unit)
+        result = await async_run_on_unit(
+            unit_name=unit, command=command, model_name=self.model_name
+        )
+        if str(result["Code"]) == "0":
+            logger.debug(result["Stdout"])
+        else:
+            logger.error(
+                "Error upgrading package on %s: %s",
+                unit,
+                result["Stderr"],
+            )
+            raise PackageUpgradeError()
+
+    async def _upgrade_packages(self) -> None:
+        """Run package updates and upgrades on each unit of an Application."""
+        status = await async_get_status()
+        app_status = status.applications.get(self.name)
+        for unit in app_status.units.keys():
+            update_command = "sudo apt update"
+            await self._run_package_upgrade_command(unit=unit, command=update_command)
+
+            dpkg_opts = "-o Dpkg::Options::=--force-confnew -o Dpkg::Options::=--force-confdef"
+            upgrade_command = f"sudo apt full-upgrade {dpkg_opts} -y"
+            await self._run_package_upgrade_command(unit=unit, command=upgrade_command)
+
+            autoremove_command = "sudo apt autoremove"
```

Comment:
`apt` is generally geared toward interactive use, not scripting, so using `apt-get` would be more appropriate here. From the man:
```
SCRIPT USAGE AND DIFFERENCES FROM OTHER APT TOOLS
       The apt(8) commandline is designed as an end-user tool and it may change behavior between versions. While it tries not to break backward compatibility this is not guaranteed either if a change seems
       beneficial for interactive use.

       All features of apt(8) are available in dedicated APT tools like apt-get(8) and apt-cache(8) as well.  apt(8) just changes the default value of some options (see apt.conf(5) and specifically the
       Binary scope). So you should prefer using these commands (potentially with some additional options enabled) in your scripts as they keep backward compatibility as much as possible.
```

Also please add `-y` flags where needed otherwise the commands would never return.

Reply:
Changed `apt` to `apt-get`. 

The previous patch already has `-y` flag for upgrade command, but I realized that autoremove needs it as well. Added.